### PR TITLE
Add version and flag logging to kube-dns.

### PR DESCRIPTION
**Pull request by mml**
Helps with #28358

Stopgap until we have a general solution for #27744

### CODE REVIEW

**Comment by mml** [comment]
```
I0701 18:33:43.800672       1 server.go:100] v1.4.0-alpha.0.1137+bf1378514920aa-dirty
I0701 18:33:43.800758       1 server.go:102] FLAG: --alsologtostderr="false"                                                                                                                  
I0701 18:33:43.800786       1 server.go:102] FLAG: --dns-port="10053"                                                                                                                         
I0701 18:33:43.800849       1 server.go:102] FLAG: --domain="cluster.local."                                                                                                                  
I0701 18:33:43.800874       1 server.go:102] FLAG: --federations="federation=kube.5yetis.net"
I0701 18:33:43.800926       1 server.go:102] FLAG: --healthz-port="8081"
I0701 18:33:43.800937       1 server.go:102] FLAG: --kube-master-url=""
I0701 18:33:43.800947       1 server.go:102] FLAG: --kubecfg-file=""                                                                                                                          
I0701 18:33:43.800960       1 server.go:102] FLAG: --log-backtrace-at=":0"                                                                                                                    
I0701 18:33:43.800970       1 server.go:102] FLAG: --log-dir=""                                                                                                                               
I0701 18:33:43.801011       1 server.go:102] FLAG: --log-flush-frequency="5s"
I0701 18:33:43.801026       1 server.go:102] FLAG: --logtostderr="true"
I0701 18:33:43.801035       1 server.go:102] FLAG: --stderrthreshold="2"
I0701 18:33:43.801043       1 server.go:102] FLAG: --v="0"                                                                                                                                    
I0701 18:33:43.801096       1 server.go:102] FLAG: --version="false"                                                                                                                          
I0701 18:33:43.801104       1 server.go:102] FLAG: --vmodule=""                                                                                                                               
```

**Comment by mml** [comment]
And a reminder that we will need to cut a new kubedns docker image before this will be run.

File: cmd/kube-dns/app/server.go [hunk 1]
```diff
@@ -94,6 +97,10 @@ func newKubeClient(dnsConfig *options.KubeDNSConfig) (clientset.Interface, error
 }
 
 func (server *KubeDNSServer) Run() {
+	glog.Infof("%+v", version.Get())
+	pflag.VisitAll(func(flag *pflag.Flag) {
```
Comment (madhusudancs) [inline cmd/kube-dns/app/server.go]: Nice! TIL ...

**Comment by madhusudancs** [comment]
LGTM. Thanks!

**Pull request merged**